aptos-move/framework/aptos-stdlib/sources/math.move

```diff
@@ -0,0 +1,75 @@
+/// Standard math utilities missing in the Move Language.
+module aptos_std::math {
+
+    /// Return the largest of two numbers.
+    public fun max(a: u64, b: u64): u64 {
+        if (a >= b) a else b
+    }
+
+    /// Return the smallest of two numbers.
+    public fun min(a: u64, b: u64): u64 {
+        if (a < b) a else b
+    }
+
+    /// Return the average of two.
+    public fun average(a: u64, b: u64): u64 {
+        (a & b) + (a ^ b) / 2
```

Comment:
I wonder if we really need the gas optimization here vs just doing (a + b) / 2

Reply:
Especially as a ^ b can overflow

Reply:
Got your point

Reply:
I think `a + b` has the overflow risk other than the original solution.
let's do `(a & b) + (a ^ b) >> 1` using bit ops for the whole expression.